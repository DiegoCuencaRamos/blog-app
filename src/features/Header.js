import React from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import { startLogout } from './auth/authSlice'

export const Header = ({ startLogout }) => (
    <header className="header">
        <div className="container">
            <div className="header__content">
                <Link to="/dashboard">
                    <h1>My Blog</h1>
                </Link>
                <button 
                    className="button--link"
                    onClick={startLogout}
                >Logout</button>
            </div>
        </div>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(Header)