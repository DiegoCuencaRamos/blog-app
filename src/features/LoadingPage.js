import React from 'react';

const LoadingPage = () => (
    <div className="loader">
        <img
            className="loader__image" 
            src="scr/images/loader.gif"
        />
    </div>
);

export default LoadingPage;