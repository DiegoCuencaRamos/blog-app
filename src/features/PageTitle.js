import React from 'react'

const PageTitle = ({ title, description }) => (
    <div className="page-header">
        <div className="container">
            <h1 className="page-header__title">{title}</h1>
            <h5 className="page-header__text">{description}</h5>
            </div>
    </div>
)

export default PageTitle

