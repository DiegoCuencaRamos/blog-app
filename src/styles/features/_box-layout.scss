@use '../base/settings';

.box-layout {
    @extend %gradient-background;
    @include settings.flex(wrap, center, center);
    background-size: cover;
    height: 100vh;
    width: 100vw;
}

.box-layout__box {
    background-color: #ffffff;
    border-radius: settings.$border-radius;
    box-shadow: settings.$box-shadow;
    padding: settings.$l-size settings.$m-size;
    text-align: center;
    width: 26rem;
}

.box-layout__title {
    margin: 0 0 settings.$s-size 0;
}