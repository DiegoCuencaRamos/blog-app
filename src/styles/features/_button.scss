@use '../base/settings';

// Buttons
%default-button {
    color: #fff;
    border-width: 0px;
    border-radius: settings.$border-radius;
    display: inline-block;
    font-size: 1.7rem;
    line-height: settings.$xl-size;
    padding: 0 settings.$m-size;
    transition: all 0.3s ease-in-out;
}

.button {
    @extend %default-button;
    background-color: #000;
}

.button:hover {
    background-color: settings.$purple;
}

.button--link {
    @extend %default-button;
    background-color: initial;
}

.button--link:hover {
    background-color: initial;
    text-decoration: underline;
}

.button--remove {
    @extend %default-button;
    background-color: settings.$dark-gray;
}

.button--remove:hover {
    background-color: settings.$blue;
}

// Links
.link {
    color: deeppink;
    display: block;
    margin-bottom: settings.$m-size;
}

.link:hover {
    text-decoration: underline;
}



