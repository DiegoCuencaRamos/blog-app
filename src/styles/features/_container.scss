@use '../base/settings';

.container {
    max-width: 80rem;
    margin: 0 auto;
    padding: 0 settings.$s-size;
}