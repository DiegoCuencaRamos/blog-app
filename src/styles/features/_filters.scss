@use '../base/settings';

// Containers
.filters {
    margin-bottom: settings.$m-size;
}

.filters__wrapper {
    flex-direction: row;
    border-radius: settings.$border-radius;
    box-shadow: settings.$box-shadow;
    padding: settings.$s-size 2.8rem;
    @media screen and (min-width: settings.$tablet-breakpoint) {
        @include settings.flex(wrap, stretch, space-between);
        padding: settings.$s-size settings.$l-size;
    }
}

.filters__content {
    @include settings.flex(wrap, stretch, $justify-content: false);
}

// Items
.filters__item {
    border: 0px;
    border-bottom: 1px solid settings.$light-gray;
    display: inline-block;
    height: settings.$xl-size;
    line-height: settings.$xl-size;
    margin-bottom: settings.$s-size;
    width: 100%;
    @media screen and (min-width: settings.$tablet-breakpoint) {
        margin-bottom: 0;
        width: auto;
    }
}

.filters__item:first-child {
    @media screen and (min-width: settings.$tablet-breakpoint) {
        margin-right: settings.$s-size;
    }
}

.filters__item::placeholder,
select.filters__item {
    color: settings.$gray;
}

.filters__item:focus {
    outline: none;
}