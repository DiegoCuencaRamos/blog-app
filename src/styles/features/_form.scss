@use '../base/settings';

// Denormalize
button, input, optgroup, select, textarea {
    line-height: 2.7rem;
}

// Wrappers
.form > * {
    margin-bottom: settings.$s-size;
}

%form__item-wrapper {
    border-radius: settings.$border-radius;
    box-shadow: settings.$box-shadow;
}

.form__item-wrapper--title {
    @extend %form__item-wrapper;
    @extend %gradient-background;
    padding-bottom: 3px;
}

.form__item-wrapper--normal {
    @extend %form__item-wrapper;
}

// Elements
%form__item {
    border-top-left-radius: settings.$border-radius;
    border-top-right-radius: settings.$border-radius;
    border-width: 0;
    outline: none;
    padding: settings.$s-size settings.$m-size;
    width: 100%;
}

.form__item--normal {
    @extend %form__item;
    height: settings.$xl-size;
}

.form__item--textarea {
    @extend %form__item;
    height: 27rem;
}

textarea {
    font-size: settings.$text-small;
}

.form__buttons-wrapper .button {
    margin-right: settings.$s-size;
    margin-bottom: settings.$s-size;
}