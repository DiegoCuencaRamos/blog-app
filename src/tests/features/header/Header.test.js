import React from 'react'
import { shallow } from 'enzyme'
import { Header } from '../../../features/Header'

let startLogout, wrapper;

beforeEach(() => {
    startLogout = jest.fn()
    wrapper = shallow(<Header startLogout={startLogout}/>)
})

// Snapshot
test('should render Header component', () => {
    expect(wrapper).toMatchSnapshot()
})

// User interaction
test('should handle startLogout on button clicked', () => {
    wrapper.find('button').simulate('click')
    expect(startLogout).toHaveBeenCalled()
})