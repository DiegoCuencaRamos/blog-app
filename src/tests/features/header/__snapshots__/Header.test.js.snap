// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header component 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__content"
    >
      <Link
        to="/dashboard"
      >
        <h1>
          My Blog
        </h1>
      </Link>
      <button
        className="button--link"
        onClick={[MockFunction]}
      >
        Logout
      </button>
    </div>
  </div>
</header>
`;
